src/common/models/data-cube/data-cube.ts

```diff
@@ -851,7 +842,7 @@ export class DataCube implements Instance<DataCubeValue, DataCubeJS> {
           expression = $(name);
           if (this.getDimensionByExpression(expression)) continue;
           // Add to the start
-          dimensions = dimensions.unshift(new Dimension({
```

Comment:
Mutations are scary. Consider later refactor and immutable data-structure usage.